### PR TITLE
feat: don't charge for pubdata in Validium mode

### PR DESCRIPTION
**Pull request by Oppen**
## What ❔

Check for the `ETH_SENDER_SENDER_VALIDIUM_MODE` variable while computing the fees to avoid charging for pubdata when that is set to `true`.
Adds an integration test that prints the fees of minting and transferring a token.

## Why ❔

Users and operators expect reduced costs compared to rollup due to not storing the pubdata in the L1.

## Run
- Install `zk` tool:
    Inside `zksync-era` run
    ```
    export ZKSYNC_HOME=$(pwd)
    export PATH=$ZKSYNC_HOME/bin:$PATH
    zk
    ```
- Run `zk` tool to initiate validium:
    ```
    zk init --validium-mode
    ```
- Run example: follow `validium.md` guide.

## Debugging

```
zk && zk clean --all && zk init --validium-mode

### CODE REVIEW

**Review by ilitteri — APPROVED** [review]
LGTM

File: .gitmodules
Comment (mationorato) [inline .gitmodules:1]: I think you want to point to 
url = https://github.com/lambdaclass/era-contracts.git
branch = validium_mode

**Review by mationorato — CHANGES_REQUESTED** [review]
- please remove changes in older vms. we shoudl only modify vm_latests.
- Look at the comment about the submodule

File: validium_mode_example/Context.sol [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
```
Comment (popzxc) [inline validium_mode_example/Context.sol:1]: Once we get closer to the mergeable solution, we'd expect the proper logical structure of the project (this applies to test code as well -- I assume this package will evolve to some form of integration test).

**Review by popzxc — COMMENTED** [review]
I understand that this is a PoC phase, so please don't treat my comments as criticism of the existing solution; I understand that there will be some refactoring/rework before validium mode is added to the main codebase.

I just thought it may be helpful to elaborate on our expectations code-wise.

File: core/tests/ts-integration/src/diff_output.txt
Comment (ilitteri) [inline core/tests/ts-integration/src/diff_output.txt:1]: Let's remove this file

File: core/tests/ts-integration/src/diff_validium.txt
Comment (ilitteri) [inline core/tests/ts-integration/src/diff_validium.txt:1]: Let's remove this file

File: core/tests/ts-integration/src/rollup_output.txt
Comment (ilitteri) [inline core/tests/ts-integration/src/rollup_output.txt:1]: Let's remove this file

File: core/tests/ts-integration/src/validium_output.txt
Comment (ilitteri) [inline core/tests/ts-integration/src/validium_output.txt:1]: Let's remove this file

File: core/tests/ts-integration/src/validium_pubdata_0_output.txt
Comment (ilitteri) [inline core/tests/ts-integration/src/validium_pubdata_0_output.txt:1]: Let's remove this file

File: core/lib/multivm/src/versions/vm_latest/oracles/storage.rs
Comment (ilitteri) [inline core/lib/multivm/src/versions/vm_latest/oracles/storage.rs:1]: Let's rollback this changes

**Review by ilitteri — CHANGES_REQUESTED** [review]
Left some comments

File: core/lib/multivm/src/versions/vm_latest/oracles/storage.rs [hunk 2]
```diff
@@ -456,9 +456,17 @@ fn get_pubdata_price_bytes(initial_value: U256, final_value: U256, is_initial: b
         compress_with_best_strategy(initial_value, final_value).len() as u32;
 
     if is_initial {
-        (BYTES_PER_DERIVED_KEY as u32) + compressed_value_size
+        println!("is initial");
```
[the remark this answers is not in the document]
Comment (ilitteri) [reply core/lib/multivm/src/versions/vm_latest/oracles/storage.rs:459]: Btw, we'll remove this rn cause is not useful anymore

**Review by ilitteri — APPROVED** [review]
LGTM, we still need to update the `era-contract` branch, but this will be another PR

**Pull request merged**